src/User/UserContainer.jsx

```diff
@@ -1,53 +1,96 @@
 import React, { useState, useEffect } from "react";
-import { Switch, Route, useParams } from "react-router-dom";
+import { Switch, Route, Redirect, useParams } from "react-router-dom";
 import { withCookies, useCookies } from "react-cookie";
 import { cookieName } from "../Constants/Cookies";
 import ProfilePage from "./ProfilePage";
 import UpdateProfile from "./UpdateProfile";
 import { getUserById } from "../api.js";
 
-const UserContainer = ({ sessionToken }) => {
-	const [user, setUser] = useState(null);
-	const [isLoading, setIsLoading] = useState(true);
-	const [error, setError] = useState(false);
-	const { id } = useParams();
+const UserContainer = () => {
+  const [user, setUser] = useState(null);
+  const [isLoading, setIsLoading] = useState(true);
+  const [error, setError] = useState(false);
+  const [managerName, setManagerName] = useState("");
+  const { id } = useParams();
   const [cookies] = useCookies([cookieName]);
-    
-	const fetchData = async () => {
-		setIsLoading(true);
-
-		try {
-			const user = await getUserById(id, cookies.sessionToken.token);
-			setUser(user);
-		} catch (error) {
-			setError(true);
-		}
-		setIsLoading(false);
-	};
-
-	useEffect(() => {
-		if (id) {
-			fetchData(id);
-		}
-	}, [id]);
-
-	return (
-		<Switch>
-			<Route path="/users/:id/edit">
-				<UpdateProfile
-					user={user}
-					id={id}
-					setUser={setUser}
-					isLoading={isLoading}
-					error={error}
-					sessionToken={sessionToken}
-				/>
-			</Route>
-			<Route path="/users/:id">
-				<ProfilePage user={user} isLoading={isLoading} error={error} />
-			</Route>
-		</Switch>
-	);
+
+  const isCurrentUser = () => {
+    return Number(id) === cookies.sessionToken.userId;
+  };
+
+  const getManagerName = async user => {
+    if (user) {
+      if (user.managerId) {
+        try {
+          const manager = await getUserById(
+            user.managerId,
+            cookies.sessionToken.token
+          );
+          setManagerName(`${manager.firstName} ${manager.lastName}`);
+        } catch (error) {
+          setError(true);
+        }
+      } else {
+        setManagerName("No_Manager");
+      }
+    }
+  };
+
+  const fetchData = async userId => {
+    setIsLoading(true);
+
+    try {
+      const user = await getUserById(userId, cookies.sessionToken.token);
+      setUser(user);
+      setError(false);
+    } catch (error) {
+      setError(true);
+    }
+
+    setIsLoading(false);
+  };
+
+  useEffect(() => {
+    if (id) {
+      fetchData(id);
+    }
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [id]);
+
+  useEffect(() => {
+    if (isCurrentUser) {
+      getManagerName(user);
+    }
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [isCurrentUser, user]);
```

Comment:
im not sure that this will work, the useEffect triggers when the _value_ changes, this will basically always check if `function isCurrentUser` === `function isCurrentUser`, which is probably always false as you are redefining it?

you might be able to do
```
useEffect(() => {
    ...
  }, [isCurrentUser(), user]);
```

The fact that you are having to use lint disable is a hint you are doing something wrong!

Reply:
I think we fixed the triggering issues but without the lint I get three warnings now:
`Line 67:6:   React Hook useEffect has missing dependencies: 'getManagerName', 'isCurrentUser', and 'user'. Either include them or remove the dependency array  react-hooks/exhaustive-deps`
` Line 67:7:   React Hook useEffect has a complex expression in the dependency array. Extract it to a separate variable so it can be statically checked          react-hooks/exhaustive-deps`
`Line 67:24:  React Hook useEffect has a complex expression in the dependency array. Extract it to a separate variable so it can be statically checked          react-hooks/exhaustive-deps`
I can't really see how to fix these as they seem to be similar warnings as the ones we get for the fetch data useEffect and every time I try to do what it asks it breaks something else?

Reply:
I tried using the `useCallback` solution but that doesn't actually deal with the `getManagerName` missing dependencies thing, it actually doesn't seem to be doing anything different really.

Reply:
I'm going to merge this now (since this technically does work and it's a warning not an error which looks like it might take a while to sort out) because we need to get the feature out before our demo, but I have written down the issue and will make it into it's own ticket to be done post feature freeze.